### PR TITLE
fix(interactive-chart): invalid legend render when add new seriesList

### PR DESCRIPTION
**Pull request by Nantawat-Poothong**
## Description
After add new seriesList to no data interactive-chart, a legend show invalid render such as `[object Object]`

### Root cause
`hasDataPoint` value is using in condition to render a legend but not update due to latest data. 

### Solution
update `hasDataPoint` when update legend with latest data.

Fixes # (issue)
https://jira.refinitiv.com/browse/ELF-2276

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: packages/elements/src/interactive-chart/__test__/interactive-chart.test.js [hunk 1]
```diff
@@ -534,5 +534,36 @@ describe('interactive-chart/InteractiveChart', function () {
...
+            symbol: 'Price',
+            type: 'candlestick'
+          }
+        ]
+      };
+      await elementUpdated(el);
+      await nextFrame(3); // wait for resize observer & rendering completion
+      expect(el.hasDataPoint).to.be.false;
+
+      await nextFrame();
```
Comment (Theeraphat-Sorasetsakul) [inline packages/elements/src/interactive-chart/__test__/interactive-chart.test.js]:
```suggestion
```

File: packages/elements/src/interactive-chart/__test__/interactive-chart.test.js [hunk 2]
```diff
@@ -534,5 +534,36 @@ describe('interactive-chart/InteractiveChart', function () {
...
+      el.seriesList[0].setData(data);
+      el.config.series[0].data = data;
+      await elementUpdated(el);
+      await nextFrame(3); // wait for resize observer & rendering completion
+      expect(el.hasDataPoint).to.be.true;
```
Comment (Theeraphat-Sorasetsakul) [inline packages/elements/src/interactive-chart/__test__/interactive-chart.test.js:568]: I think it's better to add expect point on rowLegend. Like this.
```javascript
expect($0.rowLegend[0].querySelector('.price').textContent).to.not.equal('N/A');
```
What do you think?

Comment (Nantawat-Poothong) [reply packages/elements/src/interactive-chart/__test__/interactive-chart.test.js:568]: I don't like html manual query method hahaha. check `hasDataPoint` is more specific to the root cause I think.
moreover I think it should compare with 'object Object' not 'N/A'

**Review by Theeraphat-Sorasetsakul — DISMISSED** [review]

File: packages/elements/src/interactive-chart/__test__/interactive-chart.test.js [hunk 3]
```diff
@@ -534,5 +534,38 @@ describe('interactive-chart/InteractiveChart', function () {
...
+      await nextFrame(3); // wait for resize observer & rendering completion
+      expect(el.hasDataPoint).to.be.false;
+
+      el.seriesList[0].setData(data);
+      el.config.series[0].data = data;
+      await elementUpdated(el);
+      await nextFrame(3); // wait for resize observer & rendering completion
+      expect(el.hasDataPoint).to.be.true;
+      const legendText = el.rowLegend[0].textContent;
+      const isIncludedPrices = [5679, 5694, 5544, 5547].every((price) => legendText.includes(price));
```
Comment (Theeraphat-Sorasetsakul) [inline packages/elements/src/interactive-chart/__test__/interactive-chart.test.js]:
```suggestion
      const { open, high, low, close } = data[0];
      const isIncludedPrices = [open, high, low, close].every((price) => legendText.includes(price));
```

**Review by Theeraphat-Sorasetsakul — DISMISSED** [review]

**Review by Theeraphat-Sorasetsakul — APPROVED** [review]

**Comment by bualoy-napat** [comment]
LGTM!

**Pull request merged**